src/Uno.UWP/UI/Input/Preview.Injection/InputInjector.cs

```diff
@@ -68,6 +68,8 @@ public void InjectTouchInput(IEnumerable<InjectedInputTouchInfo> input)
 		var touch = _touch!.Value.state;
 		foreach (var info in input)
 		{
+			touch.Timestamp = (ulong)DateTime.Now.Ticks; // Timestamp is stale
```

Comment:
@dr1rrb

Reply:
This however is invalid, the `touch` represent the "current state" of the injected touch input, i.e. the info extracted from the last injected touch pointer. It will be updated on line 81 by the `touch.Update(args);`, you should not update it "before" raising the event.

Also for injected pointers, the time is virtual and controlled by the user using the [`TimeOffsetInMilliseconds` and `PerformanceCount`](https://github.com/unoplatform/uno/blob/master/src/Uno.UWP/UI/Input/Preview.Injection/InjectedInputPointerInfo.cs#L19). Using the "real time" is invalid.

Looking at what you did, I suspect the issue is here https://github.com/unoplatform/uno/blob/master/src/Uno.UWP/UI/Input/Preview.Injection/InjectedInputPointerInfo.cs#L84. I think that we should invert and use the `PerformanceCount` for the `FrameId` and the `TimeOffsetInMilliseconds * 1000` for the `Timestamp`.